packages/service-library/src/servicelib/aiohttp/application_setup.py

```diff
@@ -13,6 +13,13 @@
 APP_SETUP_KEY = f"{__name__ }.setup"
 
 
+class _SetupFunc(Protocol):
```

Comment:
Protocol... that's a nice one...  it acts like an abstract class right? what is the advantage over it?
And what is the advantage over creating the following instead:
```python
SetupFun = Callable[[web.Application, ...], bool]
```

Reply:
by inheriting from ``Prototol`` i am using what is denoted as [structural subtyping](https://www.python.org/dev/peps/pep-0544/) by saying what you write *plus* the fact that it will have a member call ``__name__``

This is an approach that allows describing duck typing to the static type checker